### ISSUE TITLE
No Posts Generated

### ISSUE THREAD

**Issue by nodeGarden** [hexojs/hexo#2106]
## Environment Info

hexo -v

```
hexo: 3.2.2
hexo-cli: 1.0.2
os: Darwin 15.6.0 darwin x64
http_parser: 2.7.0
node: 6.3.0
v8: 5.0.71.52
uv: 1.9.1
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 57.1
modules: 48
openssl: 1.0.2h
```

Your site `_config.yml`

```
...
```

Your theme `_config.yml`

```
# Logo
logo: css/images/logo.png # Empty this to unset logo and show site title in the same place

# Header
menu:
  Home: .
  Archives: archives
  Categories: categories
  Tags: tags
  About: about

# Content
excerpt_link: Read More
fancybox: true

# Profile
profile: true # Whether to show profile bar
gravatar: false # Whether to use gravatar with the email. If you do, put your email address in your site's `_config.yml`.

# Sidebar
sidebar: right # set to false if you don't want a sidebar
widgets:
- recent_posts
- category
- tag
- tagcloud
- archive
thumbnail: true

# Contacts
contacts:
  github: http://github.com/ppoffice/hexo-theme-icarus
  twitter: '#'
  facebook: '#'
  dribbble: '#'
  rss: atom.xml

# Links
links:
  Hexo: http://hexo.io

# CDN
cdn: # You can choose "useso" instead of "google apis"(default).
# cdn: useso
# OR
# cdn:

# Miscellaneous
google_site_verification:
google_analytics:
favicon: favicon.png
twitter:
google_plus:
fb_admins:
fb_app_id:
```

Plugin version(`npm ls --depth 0`)

```
├── aws-sdk@2.0.31 extraneous
├── hexo@3.2.2
├── hexo-admin@1.1.2
├── hexo-deployer-sync-s3@0.0.1 (git+https://github.com/nodeGarden/hexo-deployer-sync-s3.git#cdfebb3a0ecdf013e6b8c4e1ec822cbcc987af9a)
├── hexo-generator-archive@0.1.4
├── hexo-generator-category@0.1.3
├── hexo-generator-index@0.1.3
├── hexo-generator-tag@0.1.2
├── hexo-renderer-ejs@0.1.1
├── hexo-renderer-marked@0.2.11
├── hexo-renderer-stylus@0.3.1
└── hexo-server@0.1.3
```
## For BUG
- BUG description:

I have tried `hexo clean` and `hexo generate` multiple times, but no Posts are generated. I have confirmed that everything is in the `source/_posts` folder, and even created a new draft then did `hexo publish <draftName>`, and it successfully moved. However when it goes through the generation it says it's **processed** the `.md` file, and even generates the images from within the posts, but never any `.html` files.
## log with `hexo generate --debug`

```
04:31:04.073 DEBUG Hexo version: 3.2.2
04:31:04.078 DEBUG Working directory: ~/<..directory..>/
04:31:04.268 DEBUG Config loaded: ~/<..directory..>/_config.yml
04:31:04.547 DEBUG Plugin loaded: hexo-deployer-sync-s3
04:31:04.549 DEBUG Script loaded: themes/icarus/scripts/fancybox.js
04:31:04.551 DEBUG Loading database.
04:31:04.564 INFO  Start processing
04:31:04.686 DEBUG Processed: categories/index.md
04:31:04.687 DEBUG Processed: about/index.md
04:31:04.689 DEBUG Processed: images/3b1ff3b_128x128.jpg
04:31:04.692 DEBUG Processed: images/3b1ff3b_158x158.jpg
04:31:04.693 DEBUG Processed: images/3b1ff3b_158x158.png
04:31:04.694 DEBUG Processed: images/3b1ff3b_200x300.jpg
04:31:04.695 DEBUG Processed: _posts/hexo-deployer.md
04:31:04.695 DEBUG Processed: tags/index.md
04:31:04.696 DEBUG Processed: _posts/android-to-ios.md
...
04:31:04.802 DEBUG Theme config loaded.
04:31:04.803 DEBUG Processed: _config.yml
04:31:04.804 DEBUG Processed: _posts/new-blog-with-hexo/yaml.png
04:31:04.806 DEBUG Processed: _posts/skills-bars/Screen-Shot-2015-02-24-at-4.59.13-PM.png
04:31:04.808 DEBUG Processed: source/css/style.styl
04:31:04.812 DEBUG Processed: source/js/script.js
04:31:04.812 DEBUG Processed: source/fancybox/fancybox_loading.gif
...
04:31:05.042 DEBUG Generator: page
04:31:05.042 DEBUG Generator: post
04:31:05.044 DEBUG Generator: asset
04:31:05.047 INFO  Files loaded in 496 ms
04:31:05.133 INFO  Generated: css/overwrite.css
04:31:05.142 INFO  Generated: about/index.md
04:31:05.143 INFO  Generated: images/3b1ff3b_128x128.jpg
04:31:05.143 INFO  Generated: categories/index.md
...
04:31:05.188 INFO  39 files generated in 141 ms
04:31:05.217 DEBUG Database saved
```

**Comment by NoahDragon** [comment]
Could please paste the Hexo _config.yml file?

**Comment by nodeGarden** [comment]
So funny enough I've run into this exact problem again. Sadly I didn't document how I solved last year... and I should have closed the issue when I did solve.

New versions:
```bash
hexo: 3.3.1
hexo-cli: 1.0.2
os: Darwin 16.5.0 darwin x64
http_parser: 2.7.0
node: 7.5.0
v8: 5.4.500.48
uv: 1.10.2
zlib: 1.2.8
ares: 1.10.1-DEV
modules: 51
openssl: 1.0.2k
icu: 58.2
unicode: 9.0
cldr: 30.0.3
tz: 2016j
```

_config.yml
```yaml
# Hexo Configuration
## Docs: http://hexo.io/docs/configuration.html
## Source: https://github.com/hexojs/hexo/

# Site
title: SITENAME.com v.0.0.3
subtitle:
description: My own personal blog, with Ideas, Code Snippets, Geek Rants, Foodie pictures, Travel tips, and other random things.
language: en
timezone: America/Chicago
author: Armando Rosario
author_title: 'Creative Problem Solver / Photographer / Foodie / Geek'
avatar: /images/IMG_0263_edited.jpg
location: 'Chicago, IL'
email: mondo@SITENAME.com
since: 2015 # The start year showing in your copyright section.
customstylesheet: overwrite.css

# URL
## If your site is put in a subdirectory, set url as 'http://yoursite.com/child' and root as '/child/'
url: http://SITENAME.com
root: /
permalink: :category/:title/
permalink_defaults:

# Directory
source_dir: source
public_dir: public
tag_dir: tags
archive_dir: archives
category_dir: categories
code_dir: downloads/code
i18n_dir: :lang
skip_render:
full_path: /Users/SITENAME/Desktop/S3_Mondo/Projects/SITENAME.com

# Writing
new_post_name: :title.md # File name of new posts
default_layout: post
titlecase: false # Transform title into titlecase
external_link: true # Open external links in new tab
filename_case: 1
render_drafts: false
post_asset_folder: true
relative_link: false
future: true
highlight:
  enable: true
  line_number: true
  tab_replace:

# Category & Tag
default_category: random
category_map:
tag_map:

# Date / Time format
## Hexo uses Moment.js to parse and display date
## You can customize the date format as defined in
## http://momentjs.com/docs/#/displaying/format/
date_format: YYYY-MM-DD
time_format: HH:mm:ss

# Pagination
## Set per_page to 0 to disable pagination
per_page: 6
pagination_dir: page

# Extensions
## Plugins: https://github.com/hexojs/hexo/wiki/Plugins
## Themes: https://github.com/hexojs/hexo/wiki/Themes
theme: icarus
plugins:
- hexo-deployer-sync-s3

# Disqus
disqus_shortname: SITENAMEcom

# DuoShuo
# duoshuo_shortname: hexo-theme-icarus

# Swiftype
# swiftype_install_key: XXX_XXXXXXXXXXXXXXXX

# Deployment
## Docs: http://hexo.io/docs/deployment.html
deploy:
  type: s3
  bucket: SITENAME.com
  aws_key: XXXXXXXXX
  aws_secret: XXX_XXXXXXXXXXXXXXXX
  concurrency: 8
  region: us-east-1






# Header
menu:
  Home: .
  About: about
  Categories: categories
  Tags: tags
  Archives: archives

# Content
excerpt_link: Read More >>
fancybox: true

# Sidebar
sidebar: right
widgets:
- recent_posts
- category
- tag
- archive
thumbnail: false

# Contacts
contacts:
  linkedin: https://www.linkedin.com/in/SITENAME
  github: http://github.com/SITENAME/
  twitter: '#SITENAME'
  rss: atom.xml

# Links
links:
  SITENAME: http://www.SITENAME.com/

# Miscellaneous
google_analytics:
favicon: favicon.png
twitter:
google_plus:
fb_admins:
fb_app_id:

```

Basically the same output. With the `--debug` flag on `generate`, it says its calling the `post`, `page`, and `asset` generators, but only assets are generated, and not even all of them... it always stops within the fontawesome directory:

```bash
...
22:22:34.010 DEBUG Generator: page
22:22:34.011 DEBUG Generator: post
22:22:34.014 DEBUG Generator: asset
...
```

and 

```bash
...
22:22:34.116 INFO  Generated: images/3b1ff3b_500x500.png
22:22:34.118 INFO  Generated: vendor/font-awesome/fonts/fontawesome-webfont.ttf
22:22:34.119 INFO  Generated: font-awesome/fonts/fontawesome-webfont.svg
22:22:34.119 INFO  Generated: vendor/font-awesome/fonts/fontawesome-webfont.svg
22:22:34.119 INFO  110 files generated in 102 ms
22:22:34.136 DEBUG Database saved
```

**Comment by nodeGarden** [comment]
I've done NPM install and update a number of times as well, with the same result.

**Comment by nodeGarden** [comment]
@NoahDragon yes. I installed using `npm install -g hexo`, `npm install -g hexo-cli`, removing each before each install, and also trying with both installed together. One thing I noticed is that the generic instructions mention `hexo server`, but that doesn't work for me, even when I manually install `npm install -g hexo-server`, and combinations of `hexo`/`hexo-cli` with `hexo-server`. I noticed also that every time i'd install, even though I put `package.json` -> `"hexo" : "3.3.2"`, it changes to `3.3.1`.

```bash
localhost$ hexo -v
hexo: 3.3.1
hexo-cli: 1.0.2
```

I also attempted with 3 different themes, thinking it was some bug specific to the theme.

Is there perhaps a "very verbose" `--debug` flag?
